### PR TITLE
Add types to queries

### LINKED ISSUE #5: Add types to traces and spans
Use zod to validate fetched data and type it.

### PR DESCRIPTION
**Pull request by abefernan**
Closes #5.

The wip rendering of the tags also shows now as a map, since we use that type for that field now:
type-traces

### CODE REVIEW

**Review by hashedone — COMMENTED** [review]
Before approve I want to double check if I understand this correctly. You moved from generic typescript `any` assuming the received span data is a Json type as you expected to some `zod` crate which allows you to operate on typed Json validated against a schema, right?

**Comment by abefernan** [comment]
> Before approve I want to double check if I understand this correctly. You moved from generic typescript `any` assuming the received span data is a Json type as you expected to some `zod` crate which allows you to operate on typed Json validated against a schema, right?

Exactly, it runtime checks that what we get from the API is what we want, that way.

It also transforms the received data, so for example we can check the parent span with the `parentSpanId` field, instead of having to check the references array for a `CHILD_OF` element.

https://github.com/confio/cosmwasm-tracing-ui/blob/type-traces/types/txs.ts#L28-L32

**Review by hashedone — DISMISSED** [review]
LGTM

**Pull request merged**
